### PR TITLE
Single product rate

### PR DESCRIPTION
**Pull request by nickpell**
Enforces the constraint that each software plan version has exactly one product rate, and simplifies some code in the process.

### CODE REVIEW

File: corehq/apps/accounting/management/commands/cchq_software_plan_bootstrap.py [hunk 1]
```diff
@@ -167,7 +168,13 @@ def ensure_plans(dry_run, verbose, for_tests, apps):
...
                     for product_rate in product_rates:
                         product_rate.save()
-                        software_plan_version.product_rates.add(product_rate)
+                        if hasattr(SoftwarePlanVersion, 'product_rates'):
+                            software_plan_version.product_rates.add(product_rate)
+                        elif hasattr(SoftwarePlanVersion, 'product_rate'):
+                            assert len(product_rates) == 1
+                            software_plan_version.product_rate = product_rate
+                        else:
+                            raise AccountingError('SoftwarePlanVersion does not have product_rate or product_rates field')
```
Comment (dimagimon) [inline corehq/apps/accounting/management/commands/cchq_software_plan_bootstrap.py]: E501 line too long (122 > 115 characters)

File: corehq/apps/accounting/management/commands/cchq_software_plan_bootstrap.py [hunk 2]
```diff
@@ -164,10 +165,15 @@ def ensure_plans(dry_run, verbose, for_tests, apps):
...
```
Comment (dimagimon) [inline corehq/apps/accounting/management/commands/cchq_software_plan_bootstrap.py]: E501 line too long (122 > 115 characters)

**Comment by nickpell** [comment]
@benrudolph @proteusvacuum cc: @dannyroberts

File: corehq/apps/accounting/management/commands/cchq_software_plan_bootstrap.py [hunk 3]
```diff
@@ -164,10 +165,25 @@ def ensure_plans(dry_run, verbose, for_tests, apps):
...
```
Comment (proteusvacuum) [inline corehq/apps/accounting/management/commands/cchq_software_plan_bootstrap.py]: When would this happen? Looks like you remove `product_rates` in models.py, but I could be missing something.

**Comment by proteusvacuum** [comment]
Nice! One non-blocking question, otherwise LGTM

File: corehq/apps/accounting/management/commands/cchq_software_plan_bootstrap.py (hunk 3 above)
Comment (nickpell) [inline corehq/apps/accounting/management/commands/cchq_software_plan_bootstrap.py]: I don't have a specific idea of how the ```else``` block would be reached.  This is more of a "if a breaking change is introduced, fail loudly" situation

Comment (proteusvacuum) [inline corehq/apps/accounting/management/commands/cchq_software_plan_bootstrap.py]: Interesting. Wouldn't that be a good scenario for a test? I guess this only gets run once, so no big deal, but it gets confusing if there is code that you think will never be reached. 

I was actually more wondering about why you are checking if there is either `product_rate` or `product_rates`. Aren't you removing `product_rates`, so this model shouldn't have that attribute anymore?

Comment (nickpell) [inline corehq/apps/accounting/management/commands/cchq_software_plan_bootstrap.py]: The reason for this is when migrations run, this function is called at a point where the schema contains ```product_rates``` (0003).  However, for tests, this function is called when the schema contains ```product_rate```.  This ensures that the proper schema is referenced when assign a software product rate.

**Comment by nickpell** [comment]
:white_check_mark:

File: corehq/apps/accounting/management/commands/cchq_software_plan_bootstrap.py (hunk 3 above)
Comment (proteusvacuum) [inline corehq/apps/accounting/management/commands/cchq_software_plan_bootstrap.py]: Ah! I see. Thanks for the clarification.

**Pull request merged**